src/lib/plugins/login.js

```diff
@@ -1,28 +1,26 @@
 /* global BigInt */
 const Vec3 = require('vec3').Vec3
 
-const path = require('path')
 const crypto = require('crypto')
-const requireIndex = require('../requireindex')
-const plugins = requireIndex(path.join(__dirname, '..', 'plugins'))
 const playerDat = require('../playerDat')
 const convertInventorySlotId = require('../convertInventorySlotId')
+const plugins = require('./index')
 
 module.exports.server = function (serv, options) {
   serv._server.on('connection', client =>
     client.on('error', error => serv.emit('clientError', client, error)))
 
   serv._server.on('login', async (client) => {
-    if (client.socket.listeners('end').length === 0) return // TODO: should be fixed properly in nmp instead
+    if (client.socket?.listeners('end').length === 0) return // TODO: should be fixed properly in nmp instead
+    if (!serv.worldsReady) throw new Error('World is still preparing.')
     try {
       const player = serv.initEntity('player', null, serv.overworld, new Vec3(0, 0, 0))
       player._client = client
+      player._client.socket ??= { remoteAddress: '' }
```

Comment:
same here

Reply:
What about instead checking if it's defined where it's used ?

Reply:
I removed mocking `remoteAddress` since nothing will crash if it is `undefined`, but maybe it should be documented in some way I don't really know (so other plugins can know it can be undefined)

Reply:
where did you remove it ? I don't understand

Reply:
Yeah, sorry for now making it clear I meant this https://github.com/PrismarineJS/flying-squid/pull/633/commits/18fe14a6d295cf6da45920eb12432aa89421698f . However I removed mocking : https://github.com/PrismarineJS/flying-squid/pull/633/commits/6dff61f3868eebecbc3fabeac0623d5796aaf538 is it better?
I'm open to any option here.